cpp/ql/src/semmle/code/cpp/models/implementations/Printf.qll

```diff
@@ -69,8 +69,8 @@ class Sprintf extends FormattingFunction {
     or
     hasGlobalName("__builtin___sprintf_chk") and result = 3
     or
-    getQualifiedName() != "g_strdup_printf" and
-    getQualifiedName() != "__builtin___sprintf_chk" and
+    getName() != "g_strdup_printf" and
+    getName() != "__builtin___sprintf_chk" and
```

Comment:
Shouldn't this be
```
not hasGlobalName("g_strdup_printf") and
not hasGlobalName("__builtin___sprintf_chk") and
```
?

Reply:
I think that would give the same result, but I wanted to avoid the negation to minimise the performance risk. Given that we've already restricted the name in the charpred to come from a small set, do you agree it's equivalent?

Reply:
Ah, yes, I hadn't looked outside this predicate.